### ISSUE TITLE
GPU Usage on External-model

### ISSUE THREAD

**Issue by ftrentini** [matiasdelellis/facerecognition-external-model#4]
Hi, I know this is not the channel to send this kind of message, but I tried to use GPU to process externally and I really don't know if it's working. My server is processor only, but is processing faster than the dedicated external-model server GPU powered I built just to face recognition. Is there a way to really know if the model is using GPU?

**Comment by guystreeter** [comment]
Depending on the GPU you use, there may be a utility available to observe GPU usage.
I'm interest to know what you did to try using your GPU

**Comment by ftrentini** [comment]
My dedicated server has a GTX1060 plugged on a Xeon E5-2520, with 16GB of RAM. Connected by gigabit ethernet to the webserver.
And my doubt about it is just because the gpustat command showed me the usage was only about 1-2%.

**Comment by guystreeter** [comment]
So you're just using python dlib without any changes? It's just a set of bindings to the dlib library, and if the library doesn't support your GPU, (or isn't built with cuda/whatever support), you aren't going to get GPU. Try this:
```
$ python
Python 3.10.2 (main, Jan 17 2022, 00:00:00) [GCC 11.2.1 20211203 (Red Hat 11.2.1-7)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import dlib
>>> dlib.DLIB_USE_CUDA
False
```
If it doesn't return True (you see mine reports False even though I have a GTX 980), you don't have support for it in your dlib library.

**Comment by ftrentini** [comment]
Hmmmm, enlightenment!

![5da77a0994cbe2f3631a90d63f298a0e](https://user-images.githubusercontent.com/30370155/159482311-a776eb20-e099-4cc1-8777-36449bcfaaa1.jpg)
`

**Comment by ftrentini** [comment]
Solved (But not tested yet). STEP BY STEP:

- Had to install CUDA and reinstall Nvidia drivers. I did altogether by downloading and install from [nvidia cuda toolkit site](https://developer.nvidia.com/cuda-downloads?target_os=Linux&target_arch=x86_64&Distribution=Ubuntu). Choose your system, download and run.
- cuDNN: Tricky, I had to join Nvidia's dev community. I am not a dev, but doesn't matter. Followed this page: [nvidia cuDNN libs](https://developer.nvidia.com/rdp/cudnn-download). And I followed [nvidia's instructions](https://docs.nvidia.com/deeplearning/cudnn/install-guide/index.html).

After that, I: 

- Rebuild DLIB from scratch.
- Reinstall DLIB on python.
- Rebuild pdlib from scratch

And violà: 

```
Python 3.8.10 (default, Nov 26 2021, 20:14:08) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import dlib
>>> dlib.DLIB_USE_CUDA
True
>>> 
```

and PHP

![image](https://user-images.githubusercontent.com/30370155/159505824-57d7d74b-421b-4396-8a3a-a0ea1c6fb18f.png)

Now I'm gonna start some tests. Thanks a lot @guystreeter and @matiasdelellis
